src/main/java/org/alliancegenome/curation_api/jobs/executors/GeoXrefExecutor.java

```diff
@@ -0,0 +1,120 @@
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class GeoXrefExecutor extends LoadFileExecutor {
+
+	@Inject
+	GeneService geneService;
+
+	public void execLoad(BulkLoadFileHistory bulkLoadFileHistory) throws IOException {
+
+		String url = ((BulkURLLoad) bulkLoadFileHistory.getBulkLoad()).getBulkloadUrl();
+
+		XmlMapper mapper = new XmlMapper();
+		URL src = new URL(url);
+		List<String> entrezIds = mapper.readValue(src, eSearchResult.class).getIdList().getIds();
+		
+		bulkLoadFileHistory.getBulkLoadFile().setRecordCount(entrezIds.size() + bulkLoadFileHistory.getBulkLoadFile().getRecordCount());
+		bulkLoadFileDAO.merge(bulkLoadFileHistory.getBulkLoadFile());
+
+		bulkLoadFileHistory.setCount(entrezIds.size());
+		updateHistory(bulkLoadFileHistory);
+
+		String name = bulkLoadFileHistory.getBulkLoad().getName();
+		String dataProviderName = name.substring(0, name.indexOf(" "));
+		
+		BackendBulkDataProvider dataProvider = BackendBulkDataProvider.valueOf(dataProviderName);
+
+		runLoad(bulkLoadFileHistory, dataProvider, entrezIds);
+
+		bulkLoadFileHistory.finishLoad();
+		updateHistory(bulkLoadFileHistory);
+		updateExceptions(bulkLoadFileHistory);
+	}
+
+	private void runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<String> entrezIds) {
+		ProcessDisplayHelper ph = new ProcessDisplayHelper();
+		ph.addDisplayHandler(loadProcessDisplayService);
+		if (CollectionUtils.isNotEmpty(entrezIds)) {
+			String loadMessage = "GEO cross-reference update";
+			if (dataProvider != null) {
+				loadMessage = loadMessage + " for " + dataProvider.name();
+			}
+			ph.startProcess(loadMessage, entrezIds.size());
+			
+			history.setCount(entrezIds.size());
+			updateHistory(history);
+			
+			for (String entrezId : entrezIds) {
+				try {
+					geneService.addGeoXref(entrezId, dataProvider);
+					history.incrementCompleted();
+				} catch (ObjectUpdateException e) {
+					history.incrementFailed();
+					addException(history, e.getData());
+				} catch (KnownIssueValidationException e) {
+					Log.debug(e.getMessage());
+					history.incrementSkipped();
+				} catch (Exception e) {
+					e.printStackTrace();
+					history.incrementFailed();
+					addException(history, new ObjectUpdateExceptionData(entrezId, e.getMessage(), e.getStackTrace()));
+				}
+				if (history.getErrorRate() > 0.25) {
+					Log.error("Failure Rate > 25% aborting load");
+					updateHistory(history);
+					updateExceptions(history);
+					failLoadAboveErrorRateCutoff(history);
+					return;
+				}
+				ph.progressProcess();
+			}
+			updateHistory(history);
+			updateExceptions(history);
+			ph.finishProcess();
+		}
+	}
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class eSearchResult {
+	@JacksonXmlProperty(localName="IdList")
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'=' is not followed by whitespace.
<!-- __reviewdog__:ChBiOWNiZTY3OTliOGQ0ZjdkEgpjaGVja3N0eWxl -->